submissions/m-ruslan/memory-pair-game/app.js

```diff
@@ -0,0 +1,135 @@
+// main logic of game
+document.addEventListener("DOMContentLoaded", () => {
+  let cardID, cardValue, prevCardId, prevCardValue;
+  let qOfOpenCards = 0;
+
+  startNewGame();
+
+  document.querySelector("main").addEventListener("click", (event) => {
+    let aEl = event.target.closest(".flip-container");
+    if (!aEl) return;
+    if (aEl.id === prevCardId && aEl.id !== undefined) return;
+
+    if (qOfOpenCards < 2) {
+      qOfOpenCards++;
+      cardID = aEl.id;
+      cardValue = aEl.getAttribute("value");
+
+      flip(cardID);
+
+      if (qOfOpenCards === 1) {
+        prevCardId = cardID;
+        prevCardValue = cardValue;
+      } else if (qOfOpenCards === 2) {
+        setTimeout(() => {
+          if (cardValue === prevCardValue && cardID !== prevCardId) {
+            match(cardID, prevCardId);
+          } else {
+            dontMatch(cardID, prevCardId);
+          }
+
+          qOfOpenCards = 0;
+          prevCardId = null;
+          prevValue = null;
+
+          if (qOfCardsLeft === 0) {
+            youWon();
+          }
+        }, 1000);
+      }
+    }
+  });
+});
```

Comment:
Avoid a lot of nesting if statements. not readable. 
Read about [guard expressions](https://medium.com/@iampika/javascript-functions-e66ed8871771) and [clauses](https://learningactors.com/javascript-guard-clauses-how-you-can-refactor-conditional-logic/)